students/Kate_Koehn/Lesson05/database.py

```diff
@@ -0,0 +1,148 @@
+""""
+Program to query products and customers in the rental database.
+
+"""
+import csv
+from pymongo import MongoClient
+
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def import_csv(filename):
+    """
+    function to import csv files, this function is called in the import_data() function
+    :param filename: takes a csv, removes unexpected character from first line, reads rows into dictionaries
+    :return: list of dictionaries of rows
+    """
+    with open(filename, newline="", encoding='utf-8') as file:
+        dict_list = []
+        csv_data = csv.reader(file)
+        headers = next(csv_data, None)
+        if headers[0].startswith("\ufeff"):
+            headers[0] = headers[0][1:]
+
+        for row in csv_data:
+            row_dict = {}
+            for index, column in enumerate(headers):
+                row_dict[column] = row[index]
+
+            dict_list.append(row_dict)
+
+        return dict_list
+
+
+def import_data(directory_name, product_file, customer_file, rentals_file):
+    """
+    Function to import data into three MongoDB tables
+    :param directory_name:
+    :param product_file: csv file of products with column names: product_id,description,product_type,quantity_available
+    :param customer_file: csv file of products rented by users with column names: product_id,user_id
+    :param rentals_file: csv file of customers who have rented with column names: user_id,name,address,zip_code,
+    phone_number,email
+    :return: tuple of records added to db, tuple of errors encountered while importing data
+    """
+    product_errors = 0
+    customer_errors = 0
+    rental_errors = 0
+    try:
+        products = db["products"]
+        products.insert_many(import_csv(product_file))
+    except ImportError:
+        product_errors += 1
+    try:
+        customers = db["customers"]
+        customers.insert_many(import_csv(customer_file))
+    except ImportError:
+        customer_errors += 1
+    try:
+        rentals = db["rentals"]
+        rentals.insert_many(import_csv(rentals_file))
+    except ImportError:
+        rental_errors += 1
```

Comment:
I think this doesn't actually return the number of errors per database, just whether there was an error or not. This is because you are doing insert many, and not doing a loop. What happens where there is an error with insert_many? Does it stop wherever it was? Is there anything in the ImportError that tells you how many errors there were?